cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiTest.java

```diff
@@ -249,4 +252,58 @@ public void handle(Long timerId) {
+                "log4j.logger.org.apache.zookeeper=WARN\n" +
+                "log4j.logger.org.I0Itec.zkclient=WARN\n" +
+                "log4j.logger.org.reflections=WARN";
+
+        KafkaConnectApi client = new KafkaConnectApiImpl(vertx);
+        Checkpoint async = context.checkpoint(2);
+
+        CountDownLatch latch = new CountDownLatch(1);
+        client.updateConnectorLoggers("localhost", PORT, desired)
+                .onComplete(connectorLoggers -> context.verify(() -> {
+                    client.listConnectorLoggers("localhost", PORT)
+                        .onComplete(map -> {
+                            context.verify(() -> {
+                                assertThat(map.result().get("org.apache.zookeeper").get("level"), is("WARN"));
+                                assertThat(map.result().get("org.I0Itec.zkclient").get("level"), is("WARN"));
+                                assertThat(map.result().get("org.reflections").get("level"), is("WARN"));
+                                assertThat(map.result().get("root").get("level"), is("INFO"));
+                                assertThat(map.result().get("io.debezium").get("level"), is("ERROR"));
+                                assertThat(map.result().get("unknown"), is(nullValue()));
+                                async.flag();
+                                latch.countDown();
+                            });
+                        });
+                }));
+        latch.await(30, TimeUnit.SECONDS);
+
+        desired = "log4j.rootLogger=INFO, CONSOLE\n" +
+                "log4j.logger.org.apache.zookeeper=FATAL\n" +
+                "log4j.logger.org.I0Itec.zkclient=FATAL\n" +
+                "log4j.logger.unknown=DEBUG\n" +
+                "log4j.logger.org.reflections=FATAL";
+
+        CountDownLatch latch2 = new CountDownLatch(1);
+        client.updateConnectorLoggers("localhost", PORT, desired)
+                .onComplete(connectorLoggers -> context.verify(() -> {
+                    client.listConnectorLoggers("localhost", PORT).onComplete(map -> {
+                        context.verify(() -> {
+                            assertThat(map.result().get("org.reflections").get("level"), is("FATAL"));
+                            assertThat(map.result().get("org.apache.zookeeper").get("level"), is("FATAL"));
+                            assertThat(map.result().get("org.I0Itec.zkclient").get("level"), is("FATAL"));
+                            assertThat(map.result().get("root").get("level"), is("INFO"));
+                            assertThat(map.result().get("unknown"), is(notNullValue()));
+                            assertThat(map.result().get("unknown").get("level"), is("DEBUG"));
+                            async.flag();
+                            latch2.countDown();
+                        });
+                    });
+                }));
+        latch2.await(30, TimeUnit.SECONDS);
```

Comment:
```suggestion
        Checkpoint async = context.checkpoint();
        client.updateConnectorLoggers("localhost", PORT, desired)
                .onComplete(context.succeeding())
                .compose(connectorLoggers -> client.listConnectorLoggers("localhost", PORT))
                .onComplete(map -> context.verify(() -> {
                            assertThat(map.result().get("org.reflections").get("level"), is("FATAL"));
                            assertThat(map.result().get("org.apache.zookeeper").get("level"), is("FATAL"));
                            assertThat(map.result().get("org.I0Itec.zkclient").get("level"), is("FATAL"));
                            assertThat(map.result().get("root").get("level"), is("INFO"));
                            assertThat(map.result().get("unknown"), is(notNullValue()));
                            assertThat(map.result().get("unknown").get("level"), is("DEBUG"));
                            async.flag();
                        });
                    });
                }));
```
onComplete should verify some state and then the next call should be made in a compose, and then followed by an onComplete and so on..